src/playlist-loader.js

```diff
@@ -69,7 +69,7 @@ export const resolveSegmentUris = (segment, baseUri) => {
   * master playlist with the updated media playlist merged in, or
   * null if the merge produced no change.
   */
-export const updateMaster = (master, media) => {
+export const updateMaster = (master, media, dash = false) => {
```

Comment:
It might help separate things a bit if we move the DASH specific check into dash-playlist-loader. Maybe have a method updateAndCheckForChanges that wraps both updatePlaylist and didSegmentsChange or something like that.

Might also make it easier to test the function separately.

Reply:
Added a separate function `isPlaylistUnchanged` that is passed in as `unchangedCheck` to this function. Dash passes in it's own version that does all the dash specific things and also calls `isPlaylistUnchanged` at the top.